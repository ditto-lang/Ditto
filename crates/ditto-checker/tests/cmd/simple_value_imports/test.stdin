module Test exports (..)

import Foo as F (five)

another_five : Int = F.identity(five)
