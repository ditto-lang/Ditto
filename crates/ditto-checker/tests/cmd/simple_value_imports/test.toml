bin.name = "ditto-checker-testbin-check-module"
args = ["simple_value_imports.ditto"]
fs.sandbox = true
