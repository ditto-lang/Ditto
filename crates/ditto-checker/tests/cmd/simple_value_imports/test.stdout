{
  "module_name": [
    "Test"
  ],
  "exports": {
    "types": {},
    "constructors": {},
    "values": {
      "another_five": {
        "doc_comments": [],
        "doc_position": 0,
        "value_type": {
          "type": "PrimConstructor",
          "data": "Int"
        }
      }
    }
  },
  "types": {},
  "constructors": {},
  "values": {
    "another_five": {
      "doc_comments": [],
      "name_span": {
        "start_offset": 50,
        "end_offset": 62
      },
      "expression": {
        "expression": "Call",
        "data": {
          "span": {
            "start_offset": 71,
            "end_offset": 87
          },
          "call_type": {
            "type": "PrimConstructor",
            "data": "Int"
          },
          "function": {
            "expression": "ImportedVariable",
            "data": {
              "span": {
                "start_offset": 71,
                "end_offset": 81
              },
              "variable_type": {
                "type": "Function",
                "data": {
                  "parameters": [
                    {
                      "type": "PrimConstructor",
                      "data": "Int"
                    }
                  ],
                  "return_type": {
                    "type": "PrimConstructor",
                    "data": "Int"
                  }
                }
              },
              "variable": {
                "module_name": [
                  null,
                  [
                    "Foo"
                  ]
                ],
                "value": "identity"
              }
            }
          },
          "arguments": [
            {
              "Expression": {
                "expression": "ImportedVariable",
                "data": {
                  "span": {
                    "start_offset": 82,
                    "end_offset": 86
                  },
                  "variable_type": {
                    "type": "PrimConstructor",
                    "data": "Int"
                  },
                  "variable": {
                    "module_name": [
                      null,
                      [
                        "Foo"
                      ]
                    ],
                    "value": "five"
                  }
                }
              }
            }
          ]
        }
      }
    }
  },
  "values_toposort": [
    "another_five"
  ]
}